worlds/cv64/__init__.py

```diff
@@ -0,0 +1,330 @@
+
+        # Set up the items correctly
+        for classification in item_counts:
+            for item in item_counts[classification]:
+                self.multiworld.itempool += [self.create_item(item, classification)
+                                             for _ in range(item_counts[classification][item])]
+
+    def set_rules(self) -> None:
+        active_regions = self.multiworld.get_regions(self.player)
+
+        # Set the required Special2s to the number of available bosses or crystal events returned if Special2s
+        # are not the goal.
+        if self.options.draculas_condition != self.options.draculas_condition.option_specials:
+            self.total_s2s = 0
+            self.required_s2s = 0
+
+        for region in active_regions:
+            # Place event items where they should be.
+            for loc in region.locations:
+                if loc.address is None:
+                    event_item = get_location_info(loc.name, "event")
+                    loc.place_locked_item(self.create_item(event_item, "progression"))
+                    if event_item != iname.victory:
+                        self.total_s2s += 1
+                        if self.required_s2s < self.options.bosses_required.value:
+                            self.required_s2s += 1
+
+            # Set up and connect the region's entrances.
+            connections = {}
+            rules = {}
+            if region.name == "Menu":
+                connections, rules = get_warp_entrances(self.options, self.active_warp_list, self.player)
+            else:
+                entrance_names = get_region_info(region.name, "entrances")
+                if entrance_names is not None:
+                    connections, rules = verify_entrances(self.options, entrance_names, self.active_stage_exits,
+                                                          self.player)
+            if region.name == rname.ck_main:
+                connections.update({rname.ck_drac_chamber: ename.ck_drac_door})
+                drac_rule = get_drac_rule(self.options, self.player, self.required_s2s)
+                if drac_rule is not None:
+                    rules.update(drac_rule)
+            region.add_exits(connections, rules)
+
+        # Set the completion condition
+        self.multiworld.completion_condition[self.player] = lambda state: state.has(iname.victory, self.player)
+
+    def pre_fill(self) -> None:
+        if self.starting_stage == rname.tower_of_science:
+            if self.s1s_per_warp > 3:
+                self.multiworld.local_early_items[self.player][iname.science_key2] = 1
+        elif self.starting_stage == rname.clock_tower:
+            if (self.s1s_per_warp > 2 and self.options.multi_hit_breakables.value is False) or \
+                    (self.s1s_per_warp > 8 and self.options.multi_hit_breakables.value is True):
+                self.multiworld.local_early_items[self.player][iname.clocktower_key1] = 1
```

Comment:
Oooooooo yes, this. So, in the event you need more Special1s to warp out than there are Sphere 1 locations available, AP's fill algorithm may try placing the Special1s anyway, resulting in a fill error as it cannot place enough Special1s to let you escape the starting sphere. Despite the fact that placing the stage's single key to let you progress is always also a valid solution, which really makes me suspect the fill algorithm may have a bug or an oversight. This happens most often with smaller stages with fewer checks that you cannot reach the end of without at least one key. To get around this and reduce the chances of fill errors happening, I simply force these stages' keys early to ensure the algorithm will pick them over the Special1s should they land on the starting stage slot with more Special1s than there are locations. It's not 100% foolproof because you could still end up with highly constrained Sphere 1 locations if those problem stages land on Stage 2 with Stage 1 being a small one, but it does help a lot.